docs/fundamentals/executor/containerize-executor.md

```diff
@@ -0,0 +1,168 @@
+# Dockerize your Executor
+
+Once you have understood what an `Executor` is and how it can be used inside a `Flow`, you may be interested in wrapping this Executor into a container
+so that you can isolate its dependencies and make it ready to run in the cloud or in Kubernetes.
+
+One option is to leverage {ref}`Jina Hub <hub/index>` infrastructure to make sure your Executor can run as a container.
+
+However, you can also build a Docker image yourself and use it like any other Executor. There are some requirements
+on how this image needs to be built, the main ones being:
+
+- Jina must be installed inside the image
+- The Jina CLI command to start the Executor has to be the default entrypoint
+
+## Prerequisites
+
+To be able to understand how a Container image is built for an Executor, you need an understanding of [Docker](https://docs.docker.com/) generally, of how to write 
+a [Dockerfile](https://docs.docker.com/engine/reference/builder/), and of how to build a Docker image.
+
+In addition, to reproduce the example below it is required to have Docker installed locally.
+
+
+## Installing Jina in the Docker image
+
+Jina **must** be installed inside the Docker image. This can be achieved in one of two ways:
+
+- Use a [Jina based image](https://hub.docker.com/r/jinaai/jina) as the base image in your Dockerfile.
+This will make sure that everything needed for Jina to run the Executor is installed.
+
+```dockerfile
+FROM jinaai/jina:3.0-py37-perf
+```
+
+- Simply install Jina like any other Python package. You can do this by specifying Jina in the `requirements.txt`, 
+or by including the `pip install jina` command as part of the image building process.  
+
+```dockerfile
+RUN pip install jina==3.0
+```
+
+## Setting Jina Executor CLI as entrypoint
+
+When a containerized Executor is run inside a Flow,
+under the hood Jina basically executes `docker run` with some extra arguments.
+
+This means that Jina assumes that whatever runs inside the container runs just like it would in a regular OS process. Therefore, you need to make sure that
```

Comment:
```suggestion
This means that Jina assumes that whatever runs inside the container, also runs like it would in a regular OS process. Therefore, you need to make sure that
```